aea/connections/base.py

```diff
@@ -100,22 +101,25 @@ def __init__(
         )
 
     @property
-    def loop(self) -> Optional[AbstractEventLoop]:
+    def loop(self) -> asyncio.AbstractEventLoop:
         """Get the event loop."""
-        return self._loop
-
-    @loop.setter
-    def loop(self, loop: AbstractEventLoop) -> None:
-        """
-        Set the event loop.
-
-        :param loop: the event loop.
-        :return: None
-        """
-        assert (
-            self._loop is None or not self._loop.is_running()
-        ), "Cannot set the loop while it is running."
-        self._loop = loop
+        assert asyncio.get_event_loop().is_running()
+        return asyncio.get_event_loop()
+
+    def _ensure_connected(self) -> None:  # pragma: nocover
+        """Raise exception if connection is not connected."""
+        if not self.is_connected:
+            raise ConnectionError("Connection is not connected! Connect first!")
+
+    @contextmanager
+    def _connect_context(self) -> Generator:
```

Comment:
why not make them public?

Reply:
it's only for internal use. especially _connect_context, cause it changes internal state.

Reply:
An argument in favour of public methods is that among framework users there will be some that will develop their connection, and those methods could be designed with that in mind. Anyway, I'm happy to keep them private at the moment.